lib/Date/Utility.pm

```diff
@@ -931,6 +925,31 @@ sub truncate_to_day {
     return $popular{$tepoch} // Date::Utility->new($tepoch);
 }
 
+=head2 move_to_nth_dow
+
+Returns undef or a Date::Utility object pointing to nth day of week of the current month.
+$dow is a three letter abbreviation of weekday names.
+
+=cut
+
+sub move_to_nth_dow {
+    my ($self, $nth, $dow_abb) = @_;
+    use Time::Local qw/timegm/;
+
+    $dow_abb //= 'undef';    # For nicer error reporting below.
+
+    my $dow = $day_abbrev_to_num{lc $dow_abb} or croak 'Invalid day of week. We got [' . $dow_abb . ']';
+
+    my $time     = timegm(0, 0, 0, 1, $self->month - 1, $self->year - 1900);
+    my $_dow     = (gmtime $time)[6];                                          # 0 - Sun .. 6 - Sat
+    my $days_add = ($dow + 7 - $_dow) % 7 + ($nth - 1) * 7;
+
+    my $result = Date::Utility->new($time + 24 * 3600 * $days_add);
+
+    return unless $result->month == $self->month;
+    return $result;
+}
```

Comment:
Perhaps, now that this function is part of this module it should also use the utilities provided by this module. Meaning no `Time::Local`, no `timegm` and no `gmtime`